tests/integration/test_api_consistency.py

```diff
@@ -0,0 +1,193 @@
+import pytest
+import requests
+from src.Constants import DEFAULT_NETWORK_CONFIG_MAP, PUBLIC_NODE_URL, RewardsType
+from tests.utils import Constants
+
+# Block APIs
+from src.tzkt.tzkt_block_api import TzKTBlockApiImpl
+from src.tzstats.tzstats_block_api import TzStatsBlockApiImpl
+from src.rpc.rpc_block_api import RpcBlockApiImpl
+
+# Reward APIs
+from src.tzkt.tzkt_reward_api import TzKTRewardApiImpl
+from src.tzstats.tzstats_reward_api import TzStatsRewardApiImpl
+from src.rpc.rpc_reward_api import RpcRewardApiImpl
+
+NORMAL_TEZOS_ADDRESS = Constants.NORMAL_TEZOS_ADDRESS
+STAKENOW_ADDRESS = Constants.STAKENOW_ADDRESS
+BAKEXTZ4ME_ADDRESS = Constants.BAKEXTZ4ME_ADDRESS
+
+# These tests should not be mocked but test the overall consistency
+# accross all tezos APIs which are available in TRD
+
+
+@pytest.fixture
+def address_block_api_tzkt():
+    return TzKTBlockApiImpl(DEFAULT_NETWORK_CONFIG_MAP["MAINNET"])
+
+
+@pytest.fixture
+def address_block_api_tzstats():
+    return TzStatsBlockApiImpl(DEFAULT_NETWORK_CONFIG_MAP["MAINNET"])
+
+
+@pytest.fixture
+def address_block_api_rpc():
+    return RpcBlockApiImpl(
+        DEFAULT_NETWORK_CONFIG_MAP["MAINNET"], PUBLIC_NODE_URL["MAINNET"]
+    )
+
+
+@pytest.fixture
+def current_cycle():
+    tip = "https://api.tzstats.com/explorer/tip"
+    resp = requests.get(tip, timeout=5)
+    return int(resp.json()["cycle"])
+
+
+def test_get_revelation(
+    address_block_api_tzkt, address_block_api_tzstats, address_block_api_rpc
+):
+    assert address_block_api_tzkt.get_revelation(
+        NORMAL_TEZOS_ADDRESS
+    ) == address_block_api_tzstats.get_revelation(NORMAL_TEZOS_ADDRESS)
+    assert address_block_api_tzkt.get_revelation(
+        NORMAL_TEZOS_ADDRESS
+    ) == address_block_api_rpc.get_revelation(NORMAL_TEZOS_ADDRESS)
+    assert address_block_api_tzstats.get_revelation(
```

Comment:
Is this a duplicate test as:
a = b
a = c
b = c

I.e if a == b and a == c, b must == c
And you're already checking that each revelation is working.

Reply:
Just a redundant plausibility check :). You are right. I would still keep it just in case for some flakiness behavior.

Reply:
They need to check the reaso in the logs... I bet on RPC timeouts